data-prepper-plugins/date-processor/src/main/java/com/amazon/dataprepper/plugins/processor/date/DateProcessorConfig.java

```diff
@@ -133,10 +144,20 @@ boolean isValidMatch() {
         return true;
     }
 
-    @AssertTrue(message = "Invalid timezone provided.")
-    boolean isTimezoneValid() {
+    @AssertTrue(message = "Invalid source_timezone provided.")
+    boolean isSourceTimezoneValid() {
+        try {
+            sourceZoneId = buildZoneId(sourceTimezone);
+            return true;
+        } catch (Exception e) {
+            return false;
+        }
+    }
```

Comment:
These validation functions have side effects that beyond validating the object. This is a code smell. We should reconsider how we are constructing and validating these objects. I think this is a larger systemic issue beyond these PR. Nothing to address is this PR. I just wanted to call this out. I will put something together and create an issue for this.